charts/atlantis/templates/ingress.yaml

```diff
@@ -1,12 +1,16 @@
 {{- if .Values.ingress.enabled -}}
-{{- $apiVersion := "extensions/v1beta1" -}}
-{{- if .Values.ingress.apiVersion -}}
-{{- $apiVersion = .Values.ingress.apiVersion -}}
-{{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
-{{- $apiVersion = "networking.k8s.io/v1" -}}
-{{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta/Ingress" -}}
-{{- $apiVersion = "networking.k8s.io/v1beta" -}}
-{{- end -}}
+{{- $apiVersion := .Values.ingress.apiVersion }}
+{{- if and $apiVersion (or (eq $apiVersion "networking.k8s.io/v1") (eq $apiVersion "networking.k8s.io/v1beta1") (eq $apiVersion "extensions/v1beta1")) -}}
+{{- else -}}
+  {{- $kubeVersion := .Capabilities.KubeVersion.GitVersion }}
+  {{- if semverCompare ">=1.19-0" $kubeVersion -}}
+    {{- $apiVersion = "networking.k8s.io/v1" -}}
+  {{- else if semverCompare ">=1.14-0" $kubeVersion -}}
+    {{- $apiVersion = "networking.k8s.io/v1beta1" -}}
+  {{- else -}}
+    {{- $apiVersion = "extensions/v1beta1" -}}
+  {{- end }}
+{{- end }}
```

Comment:
This condition checks if the value of `ingress.apiVersion` is set to `networking.k8s.io/v1`, `networking.k8s.io/v1beta1`, or `extensions/v1beta1`. If the value is not set to any of these supported API versions, it sets the variable `$apiVersion` to the appropriate value based on the Kubernetes version. If the Kubernetes version is `>=1.19-0`, it sets `$apiVersion` to `networking.k8s.io/v1`. If the version is `>=1.14-0` but less than `1.19-0`, it sets `$apiVersion` to `networking.k8s.io/v1beta1`. If the version is less than `1.14-0`, it sets `$apiVersion` to `extensions/v1beta1`.